umbraco/Umbraco-CMS
# V13: Add parentkey to notifications

# Notes
- Added MoveEventInfoBase
- Added MovedToRecycleBinInfo instead, as we don't need to know the parent id/key of the recycle bin, we already know by the notification that it is moving there.
- Added parent key to move info so its passed out in notifications.
- Added FIXME's where it doesn't make sense to use the new ctor of MoveEventInfo, as those methods will be refactored anyways.

# How to test
- Test the different Notifications and ensure they have parentkeys (Copy notifications and MoveToRecycleBin)
- Lets do an example of ContentTypeMovedNotification:
- Create a document type and an empty folder in the doctype tree
- Stop Umbraco and implement the respective NotificationHandler
```
using Umbraco.Cms.Core.Composing;
using Umbraco.Cms.Core.Events;
using Umbraco.Cms.Core.Notifications;

namespace Umbraco.Cms.Web.UI;

public class MyNotificationHandler : INotificationHandler<ContentTypeMovedNotification>
{
    public void Handle(ContentTypeMovedNotification notification)
    {
        foreach (var movedEntity in notification.MoveInfoCollection)
        {
            if (movedEntity.NewParentKey is not null)
            {
                Console.WriteLine("I HAVE A PARENT KEY: " + movedEntity.NewParentKey);
            }
        }
    }
}

public class MyComposer : IComposer
{
    public void Compose(IUmbracoBuilder builder) => builder.AddNotificationHandler<ContentTypeMovedNotification, MyNotificationHandler>();
}

```
- Set a debug pointing the foreach loop
- Try to move the doctype in the empty folder and assert the movedEntity has a parent key that is not null
